### PR TITLE
Improved doxygen

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>
# 🎉 New feature

## Summary

Improved doxygen and added CI linter

## Test it

In the build directory run:

```
make doc
bash <(curl -s https://raw.githubusercontent.com/ignitionrobotics/ign-cmake/ign-cmake2/tools/doc_check.sh)
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by ahcorde** [comment]
friendly ping @jennuine

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ jobs:
         uses: ignition-tooling/action-ignition-ci@bionic
         with:
           codecov-enabled: true
+          doxygen-enabled: true
```
Comment (jennuine) [inline .github/workflows/ci.yml:17]: What will this do?

File: include/ignition/gui/Application.hh [hunk 2]
```diff
@@ -96,11 +96,11 @@ namespace ignition
       /// and plugins. This function doesn't instantiate the plugins, it just
       /// keeps them in memory and they can be applied later by either
       /// instantiating a window or several dialogs.
-      /// \param[in] _config Full path to configuration file.
+      /// \param[in] _pluginElem Full path to configuration file.
```
Comment (jennuine) [inline include/ignition/gui/Application.hh:99]: nit: Renaming to `_path` would be more clear or `_pluginPath` but this is used to load configs too. For consistency, we should also update the variable in [Application.cc](https://github.com/ignitionrobotics/ign-gui/blob/d9c78efce068db6c85cfa37d3fcadfae9ac29a24/src/Application.cc#L209)
```suggestion
      /// \param[in] _path Full path to configuration file.
```

File: tutorials/03_plugins.md [hunk 3]
```diff
@@ -119,5 +119,5 @@ Ignition GUI will look for display plugins on the following paths, in this
 order:
 
 1. All paths set on the `IGN_GUI_DISPLAY_PLUGIN_PATH` environment variable
-1. `~/.ignition/gui/display_plugins`
-1. [Display plugins that are installed with Ignition GUI](https://ignitionrobotics.org/api/gui/0.1/namespaceignition_1_1gui_1_1display_plugins.html)
+2. `~/.ignition/gui/display_plugins`
+3. [Display plugins that are installed with Ignition GUI](https://ignitionrobotics.org/api/gui/0.1/namespaceignition_1_1gui_1_1display_plugins.html)
```
Comment (jennuine) [inline tutorials/03_plugins.md:123]: Can you also update the numbers under `### Publisher`?
https://github.com/ignitionrobotics/ign-gui/blame/d9c78efce068db6c85cfa37d3fcadfae9ac29a24/tutorials/03_plugins.md#L51-L56

**Review by jennuine — COMMENTED** [review]
Looks good, some minor comments.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ahcorde) [reply .github/workflows/ci.yml:17]: This will enable a doxygen linter

**Review by jennuine — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-01-10/1228/1

**Pull request merged**